app/src/main/java/com/keylesspalace/tusky/components/timeline/viewmodel/NetworkTimelineViewModel.kt

```diff
@@ -399,23 +393,17 @@ class NetworkTimelineViewModel @Inject constructor(
         currentSource?.invalidate()
     }
 
-    private inline fun updateStatusById(
-        id: String,
-        updater: (StatusViewData.Concrete) -> StatusViewData.Concrete
-    ) {
-        val pos = statusData.indexOfFirst { it.asStatusOrNull()?.id == id }
-        if (pos == -1) return
-        updateViewDataAt(pos, updater)
-    }
-
-    private inline fun updateActionableStatusById(id: String, updater: (Status) -> Status) {
-        val pos = statusData.indexOfFirst { it.asStatusOrNull()?.id == id }
-        if (pos == -1) return
-        updateViewDataAt(pos) { vd ->
-            if (vd.status.reblog != null) {
-                vd.copy(status = vd.status.copy(reblog = updater(vd.status.reblog)))
-            } else {
-                vd.copy(status = updater(vd.status))
+    private inline fun updateStatusByActionableId(id: String, updater: (Status) -> Status) {
+        // posts can be multiple times in the timeline, e.g. once the original and once as boost
+        statusData.forEachIndexed { index, status ->
```

Comment:
hm that means scanning the whole list every time which kinda sucks but I don't have a good solution either, need some smart data structure for that index

Reply:
yeah :/

Reply:
In my head there was a hashmap which contains all toots which was updated by index. I didn't realise that you needed a full table scan 🤔